File: src/FreeBilling/Data/Entities/Address.cs
﻿using System.ComponentModel.DataAnnotations;

namespace FreeBilling.Data.Entities;

public class Address
{
  public int Id { get; set; }
  public string? AddressLine1 { get; set; }
  public string? AddressLine2 { get; set; }
  public string? AddressLine3 { get; set; }
  public string? City { get; set; }
  public string? StateProvince{ get; set; }
  public string? PostalCode { get; set; }
  public string? Country { get; set; }
}

File: src/FrontEnd/freebilling.ui/src/models/address.ts
import { z } from "zod";

export interface Address {
  id: number;
  addressLine1: string | null;
  addressLine2: string | null;
  addressLine3: string | null;
  city: string | null;
  stateProvince: string | null;
  postalCode: string | null;
  country: string | null;
}

export const AddressSchema = z.object({
  id: z.number(),
  addressLine1: z.string().min(5, "Required to be at least five characters."),
  addressLine2: z.nullable(z.string()),
  addressLine3: z.nullable(z.string()),
  city: z.string().min(1, "Required"),
  stateProvince: z.string().min(1, "Required"),
  postalCode: z.string().min(1, "Required"),
  country: z.nullable(z.string())
});

export function generateEmptyAddress() {
  return {
    id: 0,
    addressLine1: "",
    addressLine2: "",
    addressLine3: "",
    city: "",
    stateProvince: "",
    postalCode: "",
    country: ""
  }
}